packages/mock-account-provider/app/consent-screen.tsx

```diff
@@ -0,0 +1,533 @@
+  readonly contextUpdates?: { [key: string]: any }
+}
+
+class ApiSteps {
+  /*
+   * flow overview:
+   *    1. start interaction --> GET /interact/:id/:nonce
+   *    2. get grant --> GET /grant/:id/:nonce
+   *    3. user makes choice --> POST /grant/:id/:nonce/accept or /grant/:id/:nonce/reject
+   *    4. end interaction --> GET /interact/:id/:nonce/finish
+   */
+
+  public static BaseUrl = '/mock-idp/auth-proxy?hostname=localhost&port=3006'
+
+  public static async startInteraction(
+    params: Record<string, string>
+  ): Promise<ApiResponse> {
+    // start interaction --> GET /interact/:id/:nonce
+    const { interactId, nonce } = params
+    const response = await ApiSteps.apiCall(
+      'GET',
+      `/interact/${interactId}/${nonce}`
+    )
+    if (response.ok) {
+      return {
+        isFailure: false,
+        payload: {
+          interactionUrl: response.responseText
+        }
+      }
+    } else {
+      return {
+        payload: {
+          error: `status ${response.status}: ${response.responseText}`
+        },
+        isFailure: true
+      }
+    }
+  }
+
+  public static async getGrant(
+    params: Record<string, string>
+  ): Promise<ApiResponse> {
+    // get grant --> GET /grant/:id/:nonce
+    const { interactId, nonce } = params
+    const response = await ApiSteps.apiCall(
+      'GET',
+      `/grant/${interactId}/${nonce}`,
+      undefined,
+      {
+        'x-idp-secret': 'replace-me'
+      }
+    )
+    if (response.ok) {
+      const grant = JSON.parse(response.responseText)
+      return {
+        isFailure: false,
+        payload: {
+          grant
+        },
+        contextUpdates: {
+          grant
+        }
+      }
+    } else {
+      return {
+        payload: {
+          error: `status ${response.status}: ${response.responseText}`
+        },
+        isFailure: true
+      }
+    }
+  }
+
+  public static async chooseConsent(
+    params: Record<string, string>
+  ): Promise<ApiResponse> {
+    // make choice --> POST /grant/:id/:nonce/accept or /grant/:id/:nonce/reject
+    const { interactId, nonce, acceptanceDecision } = params
+    const acceptanceSubPath =
+      acceptanceDecision === 'true' ? 'accept' : 'reject'
+    const response = await ApiSteps.apiCall(
+      'POST',
+      `/grant/${interactId}/${nonce}/${acceptanceSubPath}`,
+      undefined,
+      {
+        'x-idp-secret': 'replace-me'
+      }
+    )
+    if (response.ok) {
+      return {
+        isFailure: false
+      }
+    } else {
+      return {
+        payload: {
+          error: `status ${response.status}: ${response.responseText}`
+        },
+        isFailure: true
+      }
+    }
+  }
+
+  public static async endInteraction(
+    params: Record<string, string>
+  ): Promise<ApiResponse> {
+    // end interaction --> GET /interact/:id/:nonce/finish
+    const { interactId, nonce } = params
+    const response = await ApiSteps.apiCall(
+      'GET',
+      `/interact/${interactId}/${nonce}/finish`,
```

Comment:
This endpoint returns a redirect, will this call redirect the browser if it gets a `302`?

Reply:
Yeah, the browser will redirect to whatever the `Location` header points to